editor/inspector_dock.cpp

```diff
@@ -399,6 +399,23 @@ void InspectorDock::_resource_selected(const Ref<Resource> &p_res, const String
 	EditorNode::get_singleton()->push_item(r.operator->(), p_property);
 }
 
+void InspectorDock::_files_moved(const String &p_old_file, const String &p_new_file) {
+	// Because only the file name is shown, we care about changes on the file name.
+	if (p_old_file.get_file() != p_new_file.get_file()) {
+		ObjectID current_id = EditorNode::get_singleton()->get_editor_selection_history()->get_current();
+		Object *current_obj = current_id.is_valid() ? ObjectDB::get_instance(current_id) : nullptr;
+		if (current_obj == nullptr) {
+			return;
+		}
+		Resource *res = Object::cast_to<Resource>(current_obj);
```

Comment:
```suggestion
		Ref<Resource> res(current_obj);
```
Then do `res.is_valid()`. You can skip the null check above too.